File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hmap = new HashMap<>();
        int n = nums.length;
        int[] res = new int[2];
        for(int i=0; i<n; i++) {
            if(hmap.containsKey(target - nums[i])) {
                res[0] = i;
                res[1] = hmap.get(target - nums[i]);
            }
            hmap.put(nums[i], i);
        }
        return res;
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {}
        res = [0]*2
        for i in range(len(nums)):
            search = target - nums[i]
            if search in d:
                res[0] = d[search]
                res[1] = i
            
            d[nums[i]] = i
        print(d)
        return res